pyccel/codegen/pipeline.py

```diff
@@ -179,6 +179,10 @@ def handle_error(stage):
     if language is None:
         language = 'fortran'
 
+    # Choose Cuda compiler
+    if language == 'cuda':
+        compiler = 'nvidia'
```

Comment:
We should only override user decisions if they ask for default options. `compiler` should only be changed if the argument passed is `None`

Reply:
Yes, that is why we change the default. But if a user makes a stupid decision then we should not correct them. Or at least not without raising a warning. A user could pass a different compiler to check that their arguments are being read. Is it not also possible to work with the PGI compiler?

Reply:
[ec493e8](https://github.com/pyccel/pyccel-cuda/pull/32/commits/ec493e8fc12e04a7ee41bafc866dfc24ae61e9d7)

Reply:
Why not move this slightly to avoid checking if `compiler is None` twice?

Reply:
[26deedb](https://github.com/pyccel/pyccel-cuda/pull/32/commits/26deedb585ef8dc9ba8916faaf9433af6a86f353) is this better?